Mage.Sets/src/mage/cards/j/JadeOrbOfDragonkind.java

```diff
@@ -0,0 +1,172 @@
+package mage.cards.j;
+
+import mage.MageObject;
+import mage.Mana;
+import mage.abilities.Ability;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.costs.common.TapSourceCost;
+import mage.abilities.effects.ContinuousEffect;
+import mage.abilities.effects.ContinuousEffectImpl;
+import mage.abilities.effects.ReplacementEffectImpl;
+import mage.abilities.keyword.HexproofAbility;
+import mage.abilities.mana.SimpleManaAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.counters.CounterType;
+import mage.game.Game;
+import mage.game.events.EntersTheBattlefieldEvent;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.game.stack.Spell;
+import mage.watchers.Watcher;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+/**
+ *
+ * @author NicolasCamachoP
+ */
+public final class JadeOrbOfDragonkind extends CardImpl {
+
+    public JadeOrbOfDragonkind(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ARTIFACT}, "{2}{G}");
+        
+
+        // {T}: Add {G}. When you spend this mana to cast a Dragon creature spell, it enters the battlefield with an additional +1/+1 counter on it and gains hexproof until your next turn.
+        Mana mana = Mana.GreenMana(1);
+        mana.setFlag(true);
+        SimpleManaAbility ability = new SimpleManaAbility(Zone.BATTLEFIELD, mana, new TapSourceCost());
+        ability.getEffects().get(0).setText("Add {G}. When you spend this mana to cast a Dragon creature spell," +
+                " it enters the battlefield with an additional +1/+1 counter on it and gains hexproof until your" +
+                " next turn.");
+        this.addAbility(ability);
+        // it enters the battlefield with an additional +1/+1 counter on it
+        this.addAbility(new SimpleStaticAbility(Zone.ALL, new JadeOrbOfDragonkindAdditionalCounterEffect()));
+        // gains hexproof until your next turn.
+        this.addAbility(new SimpleStaticAbility(Zone.ALL, new JadeOrbOfDragonkindHexproofEffect()), new JadeOrbOfDragonkindWatcher());
```

Comment:
I don't think that this will work properly. The watchers are reset at the end of each turn. Extend your hex proof test to check that it correctly lasts until the end of your next turn (both to see that it has the ability until then and that it looses it afterwards).

Reply:
You were right. I'm trying to use two ManaSpentDelayedTriggeredAbility but that doesn't seem to work either :(